caches/clan.go

```diff
@@ -0,0 +1,91 @@
+package caches
+
+import (
+	"fmt"
+	"math/rand"
+	"time"
+
+	gocache "github.com/patrickmn/go-cache"
+	"github.com/topfreegames/khan/models"
+)
+
+// ClansSummariesCache represents a cache for the RetrieveClansSummaries operation
+type ClansSummariesCache struct {
+	Cache                 *gocache.Cache
+	TimeToLive            time.Duration
+	TimeToLiveRandomError time.Duration
+}
+
+// GetClansSummaries returns a summary of the clans details for a given list of clans by their game
+// id and public ids
+func (c *ClansSummariesCache) GetClansSummaries(db models.DB, gameID string, publicIDs []string) ([]map[string]interface{}, error) {
+	resultMap := c.getCachedClansSummaries(gameID, publicIDs)
+	err := c.getAndCacheClansSummaries(db, gameID, resultMap)
+	if err != nil {
+		if _, ok := err.(*models.CouldNotFindAllClansError); !ok {
+			return nil, err
+		}
+	}
+	var result []map[string]interface{}
+	for _, publicID := range publicIDs {
+		if resultMap[publicID] != nil {
+			result = append(result, resultMap[publicID])
+		}
+	}
+	return result, err
```

Comment:
In fact, in this case there might be an error at this stage =(
This error: `*models.CouldNotFindAllClansError`

Reply:
Right, I missed that on the first pass. Please add this to the function documentation (the fact that it may return partial results + CouldNotFindAllClansError.)

Reply:
Done!